addons/hw_scale/controllers/main.py

```diff
@@ -338,23 +343,28 @@ def run(self):
         self.device = None
 
         while True:
-            if self.device:
-                old_weight = self.weight
-                self.read_weight()
-                if self.weight != old_weight:
-                    _logger.info('New Weight: %s, sleeping %ss', self.weight, self.protocol.newWeightDelay)
-                    time.sleep(self.protocol.newWeightDelay)
-                    if self.weight and self.protocol.autoResetWeight:
-                        self.weight = 0
+            if not self.disabled:
+                if self.device:
+                    old_weight = self.weight
+                    self.read_weight()
+                    if self.weight != old_weight:
+                        _logger.info('New Weight: %s, sleeping %ss', self.weight, self.protocol.newWeightDelay)
+                        time.sleep(self.protocol.newWeightDelay)
+                        if self.weight and self.protocol.autoResetWeight:
+                            self.weight = 0
+                    else:
+                        _logger.info('Weight: %s, sleeping %ss', self.weight, self.protocol.weightDelay)
+                        time.sleep(self.protocol.weightDelay)
+                        self.disabled = True
                 else:
-                    _logger.info('Weight: %s, sleeping %ss', self.weight, self.protocol.weightDelay)
-                    time.sleep(self.protocol.weightDelay)
-            else:
-                with self.scalelock:
-                    self.device = self.get_device()
-                if not self.device:
-                    # retry later to support "plug and play"
-                    time.sleep(10)
+                    with self.scalelock:
+                        self.device = self.get_device()
+                    if not self.device:
+                        # retry later to support "plug and play"
+                        time.sleep(10)
+                    else: 
+                        self.disabled = self.protocol.disable
+
```

Comment:
could use an extra sleep?

Reply:
(if disabled)

Reply:
no, you are looking at the wrong place ;) , this check is only for finding a device -> max 1 beep. The other checks still look at the delay values configured in the protocol.